components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -227,6 +246,40 @@ func (agent *Smith) Start(ctx context.Context, callback func(InfringingWorkspace
 		}()
 	}
 
+	egressTicker := time.NewTicker(2 * time.Minute)
+
+	go func() {
+		for {
+			<-egressTicker.C
+			for pid, ws := range workspaces {
+				// check if the workspace is already stopped
+				fi, err := os.Stat(fmt.Sprintf("/proc/%d", pid))
+				if err != nil {
+					if errors.Is(err, os.ErrNotExist) {
+						wssWriteMutex.Lock()
+						log.Debugf("deleting workspace with pid %d and workspaceId %s from workspaces", pid, ws.WorkspaceID)
+						delete(workspaces, pid)
+						wssWriteMutex.Unlock()
+					} else {
+						log.WithError(err).WithFields(log.OWI(ws.OwnerID, ws.WorkspaceID, ws.InstanceID)).Error("could not stat workspace, stale entries will continue to exist in workspaces state")
+					}
+					continue
+				}
+				infringement, err := agent.checkEgressTrafficCallback(pid, fi.ModTime())
+				if err != nil {
+					log.WithError(err).WithFields(log.OWI(ws.OwnerID, ws.WorkspaceID, ws.InstanceID)).Error("error calling checkEgressTrafficCallback")
+					continue
+				}
+				if infringement != nil {
+					// monitor metric here as we are not penalizing workspaces yet
+					log.Debugf("found infringing workspace with pid %d and worksapceId %s", pid, ws.WorkspaceID)
```

Comment:
Can you change this logging level to info (from debug)? 

That way it is similar to what we do when [stopping](https://github.com/gitpod-io/gitpod/blob/17abe734cb4252becd7a3447bb0a2b5b4ff21de9/components/ee/agent-smith/pkg/agent/agent.go#L347) or [stopping and blocking a user](https://github.com/gitpod-io/gitpod/blob/17abe734cb4252becd7a3447bb0a2b5b4ff21de9/components/ee/agent-smith/pkg/agent/agent.go#L356-L357). Why? We need to inspect the logs to determine how we want to tune the auto stoppage of workspaces in production. 

Also, just a nit, `worksapceId` should be `workspaceId`.